addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusHandler.java

```diff
@@ -0,0 +1,369 @@
+ * sent to one of the channels.
+ *
+ * @author Thomas Rokohl - Initial contribution
+ */
+public class FroniusHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusHandler.class);
+
+    private static final String InverterRealtimeDataUrl = "http://%IP%/solar_api/v1/GetInverterRealtimeData.cgi?Scope=Device&DeviceId=%DEVICEID%&DataCollection=CommonInverterData";
+    private static final String PowerFlowRealtimeData = "http://%IP%/solar_api/v1/GetPowerFlowRealtimeData.fcgi";
+    private static final int DEFAULT_REFRESH_PERIOD = 5;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private InverterRealtimeResponse inverterRealtimeResponse;
+    private PowerFlowRealtimeResponse powerFlowResponse;
+
+    private Gson gson;
+
+    public FroniusHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+        } else {
+            logger.debug("The Air Quality binding is read-only and can not handle command {}", command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Fronius handler.");
+
+        FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+        if (StringUtils.trimToNull(config.ip) == null) {
+            errorMsg = "Parameter 'Ip' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.refresh != null && config.refresh <= 0) {
+            errorMsg = "Parameter 'refresh' must be at least 1 second";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            startAutomaticRefresh();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Start the job refreshing the data
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request
+                        updateData();
+                        // Update all channels
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
```

Comment:
Instead of parsing the configuration again, how about passing its "delay value" as a parameter to `startAutomaticRefresh()`?